.jenkins/caffe2/build.sh

```diff
@@ -132,6 +132,7 @@ CMAKE_ARGS+=("-DCMAKE_INSTALL_PREFIX=${INSTALL_PREFIX}")
 
 if [[ $BUILD_ENVIRONMENT == *mkl* ]]; then
   CMAKE_ARGS+=("-DBLAS=MKL")
+  CMAKE_ARGS+=("-DUSE_MKLDNN=ON")
```

Comment:
So... I don't think this should be necessary, because we should build with MKLDNN if we can find it. It looks like the problem is that setup.py and cmake have diverged in their defaults. In setup.py I see:

```

# ppc64le and aarch64 do not support MKLDNN
if IS_PPC or IS_ARM:
    USE_MKLDNN = check_env_flag('USE_MKLDNN', 'OFF')
else:
    USE_MKLDNN = check_env_flag('USE_MKLDNN', 'ON')
```

But in CMakeLists.txt I see

```
option(USE_MKLDNN "Use MKLDNN" OFF)
```

these should be made to agree.

Reply:
@yinghai @ezyang 
I'm not sure with this.
If it's OK to build MKLDNN by default for caffe2 as well, I'll change it as above.